src/app/core/config-ui/config-field/config-field.component.ts

```diff
@@ -0,0 +1,292 @@
+    });
+
+    this.updateDataTypeAdditional(this.schemaFieldsForm.get("dataType").value);
+    this.schemaFieldsForm
+      .get("dataType")
+      .valueChanges.subscribe((v) => this.updateDataTypeAdditional(v));
+    this.updateForNewOrExistingField();
+  }
+
+  private updateForNewOrExistingField() {
+    if (!!this.fieldId) {
+      // existing fields' id is readonly
+      this.fieldIdForm.disable();
+    } else {
+      const autoGenerateSubscr = this.schemaFieldsForm
+        .get("label")
+        .valueChanges.subscribe((v) => this.autoGenerateId(v));
+      // stop updating id when user manually edits
+      this.fieldIdForm.valueChanges.subscribe(() =>
+        autoGenerateSubscr.unsubscribe(),
+      );
+    }
+  }
+  private autoGenerateId(updatedLabel: string) {
+    const generatedId = generateIdFromLabel(updatedLabel);
+    this.fieldIdForm.setValue(generatedId, { emitEvent: false });
+  }
+
+  private initAvailableDatatypes(dataTypes: DefaultDatatype[]) {
+    this.dataTypes = dataTypes
+      .filter((d) => d.label !== DefaultDatatype.label) // hide "internal" technical dataTypes that did not define a human-readable label
+      .map((d) => ({
+        label: d.label,
+        value: d.dataType,
+      }));
+  }
+  objectToLabel = (v: SimpleDropdownValue) => v?.label;
+  objectToValue = (v: SimpleDropdownValue) => v?.value;
+  createNewAdditionalOption: (input: string) => SimpleDropdownValue;
+
+  private updateDataTypeAdditional(dataType: string) {
+    this.resetAdditional();
+
+    if (dataType === ConfigurableEnumDatatype.dataType) {
+      this.initAdditionalForEnum();
+    } else if (
+      dataType === EntityDatatype.dataType ||
+      dataType === EntityArrayDatatype.dataType
+    ) {
+      this.initAdditionalForEntityRef();
+    }
+
+    // hasInnerType: [ArrayDatatype.dataType].includes(d.dataType),
+
+    // TODO: this mapping of having an "additional" schema should probably become part of Datatype classes
+  }
+
+  private initAdditionalForEnum() {
+    this.typeAdditionalOptions = this.configurableEnumService
+      .listEnums()
+      .map((x) => ({
+        label: Entity.extractEntityIdFromId(x), // TODO: add human-readable label to configurable-enum entities
+        value: Entity.extractEntityIdFromId(x),
+      }));
+    this.additionalForm.addValidators(Validators.required);
+
+    this.createNewAdditionalOption = (text) => ({
+      value: generateIdFromLabel(text),
+      label: text,
+    });
+
+    if (this.entitySchemaField.additional) {
+      this.additionalForm.setValue(this.entitySchemaField.additional);
+    } else if (this.schemaFieldsForm.get("label").value) {
+      // when switching to enum datatype in the form, if unset generate a suggested enum-id immediately
+      const newOption = this.createNewAdditionalOption(
+        this.schemaFieldsForm.get("label").value,
+      );
+      this.typeAdditionalOptions.push(newOption);
+      this.additionalForm.setValue(newOption.value);
```

Comment:
I am still not sure how problematic it is to `_id` values with a space in them. I would rather not do that as we might need to request them over API at one point.